spec/ics-1-ics-standard/README.md

```diff
@@ -0,0 +1,152 @@
+
+`required-by` - Other ICS standards, referenced by number, which require or depend upon this standard.
+
+`replaces` - Another ICS standard replaced or supplanted by this standard, if applicable.
+
+`replaced-by` - Another ICS standard which replaces or supplants this standard, if applicable.
+
+### Synopsis
+
+Following the header, an ICS should include a brief (~200 word) synopsis providing a high-level
+description of and rationale for the specification.
+
+### Specification
+
+The specification section is the main component of an ICS, and should contain protocol documentation, design rationale,
+required references, and technical details where appropriate.
+
+#### Sub-components
+
+The specification may have any or all of the following sub-components, as appropriate to the particular ICS. Included sub-components should be listed in the order specified here.
+
+- *Motivation* - A rationale for the existence of the proposed feature, or the proposed changes to an existing feature.
+- *Desiderata* - A list of the desired properties or characteristics of the protocol or feature specified.
+- *Technical Specification* - All technical details of the proposed protocol including syntax, semantics, sub-protocols, data structures, algorithms, and pseudocode as appropriate.
+    The technical specification should be detailed enough such that separate correct implementations of the specification without knowledge of each other are compatible.
+- *Backwards Compatibility* - A discussion of compatibility (or lack thereof) with previous feature or protocol versions.
+- *Forwards Compatibility* - A discussion of compatibility (or lack thereof) with future possible or expected features or protocol versions.
+- *Implementations* - A list of candidate or finalized implementations (external references, not inline).
+
+### History
+
+An ICS should include a history section, listing any inspiring documents and a plaintext log of significant changes.
+
+See an example history section [below](#history-1).
+
+### Copyright
+
+An ICS should include a copyright section waiving rights via [CC0](https://creativecommons.org/publicdomain/zero/1.0/).
+
+## Formatting
+
+### General
+
+ICS specifications must be written in GitHub-flavored Markdown.
+
+For a GitHub-flavored Markdown cheat sheet, see [here](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet). For a local Markdown renderer, see [here](https://github.com/joeyespo/grip).
+
+### Language
+
+ICS specifications should be written in Simple English, avoiding obscure terminology and unnecessary jargon. For excellent examples of Simple English, please see the [Simple English Wikipedia](https://simple.wikipedia.org/wiki/Main_Page).
+
+### Pseudocode
+
+Pseudocode in specifications should be language-agnostic and formatted in the CS-paper standard, with line numbers, variables, simple conditional blocks, for loops, and
```

Comment:
There isn't a single "CS paper standard", theres a bunch of different tex algorithm packages. Perhaps you want to link to the package / standard your considering?

Reply:
Yes, let's - also, note that we're limited to Markdown/HTML - do you have any favorites?

Reply:
I don't really have any favorites. I tend to actually dislike formats that have line numbers. This is probably recency bias, but I do like the notation in the FRI paper. https://eccc.weizmann.ac.il/report/2017/134/revision/2/download page 13

Reply:
While images allow more formatting options, they mean that diffs can't be compared, which I think is a pretty significant disadvantage. Perhaps some LaTeX could be inserted in images rendered on-demand as discussed in [this SO answer](https://stackoverflow.com/questions/11256433/how-to-show-math-equations-in-general-githubs-markdownnot-githubs-blog).

Reply:
(I guess the LaTeX source could be included and diffs of that compared, that seems OK too)

Reply:
Added that as an option.